_posts/2021-08-23-using-lra.adoc

```diff
@@ -0,0 +1,490 @@
+
+We call the interaction an LRA, short for Long Running Action. An LRA has specific properties and guarantees that aid the construction of reliable service interactions. Each action is assigned a unique identifier (referred to as the LRA context) so that it can be distinguished from other LRAs.
+
+Services start an LRA by marking a JAX-RS method with an https://download.eclipse.org/microprofile/microprofile-lra-1.0/apidocs/org/eclipse/microprofile/lra/annotation/ws/rs/LRA.html[@LRA annotation]. When such a method is invoked the system will start the action and make its identifier available as a JAX-RS header called "Long Running Action". If the body of the method performs any JAX-RS invocations the header is automatically added to outgoing requests. In this way the target services can join in with the interaction (if they are also annotated with the `@LRA` annotation).
+
+Any work performed by a method annotated in this way should be "compensatable" in the sense that if some service "cancels" the LRA then the service will be reliably notified that it should compensate for any work that it did. Each service is responsible for interpreting the notion of what it means to compensate, the only requirement is that it responds https://download.eclipse.org/microprofile/microprofile-lra-1.0/apidocs/org/eclipse/microprofile/lra/annotation/ParticipantStatus.html[appropriately] when it is notified that it should compensate (the service indicates how it should be notified by annotating one of its method with an https://download.eclipse.org/microprofile/microprofile-lra-1.0/apidocs/org/eclipse/microprofile/lra/annotation/Compensate.html[@Compensate] annotation). Refer to the javadoc for the `@LRA` annotation for the details of how to control the outcome of an LRA.
+
+IMPORTANT: The extension provides an implementation of the https://download.eclipse.org/microprofile/microprofile-lra-1.0-M1/microprofile-lra-spec.html[MicroProfile LRA specification] and its associated
+ https://download.eclipse.org/microprofile/microprofile-lra-1.0/apidocs/index.html?org/eclipse/microprofile/lra/annotation/ws/rs/LRA.html[annotations].
+
+== LRA coordinators
+
+The `narayana-lra-extension` requires the presence of a running coordinator in the environment. Coordinators can be obtained from `https://hub.docker.com/r/jbosstm/lra-coordinator`
+or you can build your own coordinator using a maven pom that includes the appropriate dependencies. For the purpose of this blog we'll show how to create from scratch using the `quarkus-maven-plugin`.
+There is some extra information about configuring the coordinator in one of the https://jbossts.blogspot.com/2021/07/narayana-lra-update.html[narayana blogs].
+
+Since the coordinator is just a JAX-RS resource we can build one using quarkus, adding the `resteasy-jackson` and `rest-client` extensions:
+
+[source,bash]
+----
+$ mvn io.quarkus:quarkus-maven-plugin:2.2.0.Final:create \
```

Comment:
Please make it 2.2.1.Final, we won't officially announce 2.2.0.Final due to a bug in dev mode on Windows.